Source/missiles.cpp

```diff
@@ -915,6 +874,105 @@ Direction16 GetDirection16(Point p1, Point p2)
 	return ret;
 }
 
+void MissileHitMonsterConsequences(int mid, int pnum, int dam, missile_id mName)
+{
+	auto &monster = Monsters[mid];
+	const auto &player = Players[pnum];
+
+	bool hasKnockback = false;
+	if (pnum >= 0) {
+		const auto &player = Players[pnum];
+		if (HasAnyOf(player._pIFlags, ItemSpecialEffect::Knockback))
+			hasKnockback = true;
+	}
+
+	if (monster._mhitpoints >> 6 <= 0) {
+		M_StartKill(mid, pnum);
+	} else if (monster.IsResistant(mName)) {
+		PlayEffect(monster, 1);
+	} else {
+		if (monster._mmode != MonsterMode::Petrified
+		    && MissilesData[mName].mType == 0
+		    && hasKnockback)
+			M_GetKnockback(mid);
+		if (monster.MType->mtype != MT_GOLEM)
+			M_StartHit(mid, pnum, dam);
+	}
+}
+
+int PlayerMissileCalculateCTHAgainstMonster(int pnum, missile_id mName, int dist, Monster &monster)
+{
+	int hper = 0;
+	if (pnum != -1) {
+		const auto &player = Players[pnum];
+		if (MissilesData[mName].mType == 0) {
+			hper = player.GetRangedPiercingToHit();
+			hper -= player.CalculateArmorPierce(monster.mArmorClass, false);
+			hper -= (dist * dist) / 2;
+		} else {
+			hper = player.GetMagicToHit() - (monster.mLevel * 2) - dist;
+		}
+	} else {
+		hper = GenerateRnd(75) - monster.mLevel * 2;
+	}
+	return hper;
+}
+
+int PlayerMissileCalculateDamageAgainstMonster(int pnum, int minDam, int maxDam, missile_id mName)
+{
+	int dam = minDam + GenerateRnd(maxDam - minDam + 1);
+
+	const auto &player = Players[pnum];

```

Comment:
Take player by ref instead of index, also function name:
```suggestion
int Missile(::)CalculateDamage(const Player &player, int minDam, int maxDam, missile_id mName)
{
	int dam = minDam + GenerateRnd(maxDam - minDam + 1);
```

Reply:
Functions parameters would be changed after merge of MonsterMHit and MonsterTrapHit together.
As for renaming part - let's wait after merge 2 functions together in following commit in this PR.
What brackets in (::) are for? I usually see this without brackets.

Reply:
(::) was to let you know of the two optional names :)

Reply:
this is probably resolved now